src/main/scala/com/codacy/parsers/implementation/JacocoParser.scala

```diff
@@ -3,59 +3,54 @@ package com.codacy.parsers.implementation
 import java.io.File
 
 import com.codacy.api._
-import com.codacy.parsers.CoverageParser
-import com.codacy.parsers.util.{TextUtils, XMLoader}
+import com.codacy.parsers.util.TextUtils
+import com.codacy.parsers.{CoverageParser, XmlReportParser}
 
-import scala.util.{Failure, Success, Try}
-import scala.xml.{Node, NodeSeq}
+import scala.xml.{Elem, Node, NodeSeq}
 
 private case class LineCoverage(missedInstructions: Int, coveredInstructions: Int)
 
-object JacocoParser extends CoverageParser {
+object JacocoParser extends CoverageParser with XmlReportParser {
 
   override val name: String = "Jacoco"
 
-  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] = {
-    val report = (Try(XMLoader.loadFile(reportFile)) match {
-      case Success(xml) if (xml \\ "report").nonEmpty =>
-        Right(xml \\ "report")
+  private val ReportTag = "report"
 
-      case Success(_) =>
-        Left("Invalid report. Could not find top level <report> tag.")
+  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] =
+    parseReportWithEither(reportFile, s"Could not find top level <$ReportTag> tag") {
+      parse(projectRoot, _)
+    }
 
-      case Failure(ex) =>
-        Left(s"Unparseable report. ${ex.getMessage}")
-    })
+  override def validateSchema(xml: Elem): Boolean = getRootNode(xml).nonEmpty
 
-    report.flatMap(parse(projectRoot, _))
-  }
+  override def getRootNode(xml: Elem): NodeSeq = xml \\ ReportTag
 
   private def parse(projectRoot: File, report: NodeSeq): Either[String, CoverageReport] = {
```

Comment:
I would give another name to this function as it has the same name as the overriden function `override def parse()`

Reply:
Done ✅